kratos/integration/integration_point_utilities.cpp

```diff
@@ -0,0 +1,1450 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+
+
+// Project includes
+#include "integration_point_utilities.h"
+
+
+namespace Kratos
+{
+    void IntegrationPointUtilities::IntegrationPoints1D(
+        typename IntegrationPointsArrayType::iterator& rIntegrationPointsBegin,
+        SizeType PointsInU,
+        double U0, double U1)
+    {
+        KRATOS_ERROR_IF(PointsInU < 1)
+            << "Degree need to be bigger than 0." << std::endl;
+
+        const double distance_u = U1 - U0;
+        const double length_u = std::abs(U1 - U0);
+
+        const std::vector<std::vector<double>> integration_point_list_u = IntegrationPointUtilities::s_gauss_legendre[PointsInU - 1];
+
+        IndexType counter = 0;
+        for (SizeType u = 0; u < PointsInU; ++u)
+        {
+            (*rIntegrationPointsBegin)[0] = U0 + distance_u * integration_point_list_u[u][0];
+            (*rIntegrationPointsBegin).Weight() =
+                integration_point_list_u[u][1] * length_u;
+
+            rIntegrationPointsBegin++;
+        }
+    }
+
+    void IntegrationPointUtilities::IntegrationPoints2D(
+        typename IntegrationPointsArrayType::iterator& rIntegrationPointsBegin,
+        SizeType PointsInU, SizeType PointsInV,
+        double U0, double U1, double V0, double V1)
+    {
+        KRATOS_ERROR_IF(PointsInU < 1 || PointsInV < 1)
+            << "Degrees need to be bigger than 0." << std::endl;
+
+        const double distance_u = U1 - U0;
+        const double length_u = std::abs(U1 - U0);
+        const double distance_v = V1 - V0;
+        const double length_v = std::abs(V1 - V0);
+
+        const std::vector<std::vector<double>> integration_point_list_u = IntegrationPointUtilities::s_gauss_legendre[PointsInU - 1];
+        const std::vector<std::vector<double>> integration_point_list_v = IntegrationPointUtilities::s_gauss_legendre[PointsInV - 1];
+
+        for (SizeType u = 0; u < PointsInU; ++u)
+        {
+            for (SizeType v = 0; v < PointsInV; ++v)
+            {
+                (*rIntegrationPointsBegin)[0] = U0 + distance_u * integration_point_list_u[u][0];
+                (*rIntegrationPointsBegin)[1] = V0 + distance_v * integration_point_list_v[v][0];
+                (*rIntegrationPointsBegin).Weight() =
+                    integration_point_list_u[u][1] * length_u *
+                    integration_point_list_v[v][1] * length_v;
+
+                rIntegrationPointsBegin++;
+            }
+        }
+    }
+
+    const std::vector<std::vector<std::vector<double>>> IntegrationPointUtilities::s_gauss_legendre = {
```

Comment:
@pooyan-dadvand when I change the inner std::vector to array_1d I cannot initialize in that easy way. Any idea about that?

Reply:
You are right! So you may use the `std::array<double,3>`